### ISSUE TITLE
Problems with building shared libraries with VS14?

### ISSUE THREAD

**Issue by jeromekelleher** [ampl/gsl#10]
I'm trying to compile GSL with shared library support using visual studio community 2015. This is to provide windows support in the conda recipe: https://github.com/conda-forge/gsl-feedstock/issues/7

The issue I'm having is as follows. I run:

```
cmake . -DBUILD_SHARED_LIB=1
msbuild GSL.sln
```

I then get an error:
```
"C:\jk-work\gsl\GSL.sln" (default target) (1) ->
"C:\jk-work\gsl\ALL_BUILD.vcxproj.metaproj" (default target) (2) ->
"C:\jk-work\gsl\block_test.vcxproj.metaproj" (default target) (4) ->
"C:\jk-work\gsl\gsl.vcxproj.metaproj" (default target) (5) ->
"C:\jk-work\gsl\gsl.vcxproj" (default target) (10) ->
(Link target) ->
  gsl.def : error LNK2001: unresolved external symbol gsl_multilarge_nlinear_df
 [C:\jk-work\gsl\gsl.vcxproj]
  gsl.def : error LNK2001: unresolved external symbol gsl_multilarge_nlinear_fd
fvv [C:\jk-work\gsl\gsl.vcxproj]
  C:/jk-work/gsl/Debug/gsl.lib : fatal error LNK1120: 2 unresolved externals [C
:\jk-work\gsl\gsl.vcxproj]

    288 Warning(s)
    3 Error(s)
```

Apologies if I'm doing something stupid here; I'm a windows noob!

**Comment by mancoast** [comment]
Greetings!

Confirmed, GIT HEAD a34d7c9e5ffebe0252c0554a17fd6c6f040e89e0 fails shared build:

 "D:\GIT\gsl\build-win\shared_GSL\gsl.sln" (default target) (1) ->
 "D:\GIT\gsl\build-win\shared_GSL\gsl.vcxproj.metaproj" (default target) (19) ->
 "D:\GIT\gsl\build-win\shared_GSL\gsl.vcxproj" (default target) (62) ->
 (Link target) ->
   gsl.def : error LNK2001: unresolved external symbol gsl_multilarge_nlinear_df [D:\GIT\gsl\build-win\shared_GSL\gsl.vcxproj]
   gsl.def : error LNK2001: unresolved external symbol gsl_multilarge_nlinear_fdfvv [D:\GIT\gsl\build-win\shared_ GSL\gsl.vcxproj]
   D:/GIT/gsl/build-win/shared_GSL/Release/gsl.lib : fatal error LNK1120: 2 unresolved externals [D:\GIT\gsl\buil d-win\shared_GSL\gsl.vcxproj]
3 Error(s)




Confirmed, GIT HEAD a34d7c9e5ffebe0252c0554a17fd6c6f040e89e0 passes static build.

cd %BUILDTREE%\static_GSL
ECHO %cmake_platform% STATIC
cmake -G %cmake_platform% ^
-DBUILD_SHARED_LIBS:BOOL=OFF ^
-DCMAKE_CXX_FLAGS_RELEASE="/MD" ^
-DCMAKE_CXX_FLAGS_DEBUG="/MDd" ^
-DCMAKE_C_FLAGS_RELEASE="/MD" ^
-DCMAKE_C_FLAGS_DEBUG="/MDd" ^
-DCMAKE_BUILD_TYPE="Release" %SRC%
msbuild gsl.sln /p:Configuration=Release /m


Thanks,
Coast

**Comment by mancoast** [comment]
resolved with commit 77c17ce8720730186281cf8ca824711437934459

**Comment by jeromekelleher** [comment]
Fantastic; thanks @mancoast!
